validator/client/runner.go

```diff
@@ -305,6 +305,32 @@ func runHealthCheckRoutine(ctx context.Context, v iface.Validator, eventsChan ch
 				return
 			}
 			isHealthy := tracker.CheckHealth(ctx)
+			for !isHealthy {
+				cSlot, err := v.CanonicalHeadSlot(ctx)
+				if err != nil {
+					log.WithError(err).Error("Could not get canonical head slot")
+				}
+				nSlot := <-v.NextSlot()
```

Comment:
Here we will have to wait if the channel is busy. I don't know if this is the best approach.